docs/specs/target.md

```diff
@@ -0,0 +1,188 @@
+[reference architectures][reference_architectures],
+[achieving high availability][high_availability], and [hardening][hardening]
+Vector.
+
+## 2. Installation Targets
+
+Vector supports a number of installation targets that can be categorized into:
+
+- Virtual/Physical Machine
+- Orchestration Platform
+
+The primary differentiator between the two being that Virtual/Physical Machines
+provide a single node as the deployment target, whereas Orchestration Platforms
+allow for a scheduler to deploy Vector across a number nodes. These categories
+have their own requirements for each
+[Deployment Architecture](#3-deployment-architectures).
+
+Examples of Virtual/Physical Machine targets include, but are not limited to:
+
+- Debian
+- Docker
+- RHEL
+- Windows
+
+Examples of Orchestration Platform targets include, but are not limited to:
+
+- Kubernetes
+
+## 3. Deployment Architectures
+
+When supporting a target, Vector must support them through the the paradigm of
+architectures:
+
+- Targets MUST support the [agent architecture][agent_architecture] by
+  providing a single command that deploys Vector and achieves the
+  [agent architecture requirements](#agent-architecture).
+- Targets SHOULD support the [aggregator architecture][aggregator_architecture] by
+  providing a single command that deploys Vector and achieves the
+  [aggregator architecture requirements](#aggregator-architecture).
+- Targets MAY support the [unified architecture][unified_architecture] by
+  providing a single command that deploys Vector and achieves the
+  [unified architecture requirements](#unified-architecture).
+
+### 4. Agent Architecture
+
+The [agent architecture][agent_architecture] deploys Vector on each individual
+node for local data collection. Along with general [hardening](#7-hardening)
+requirements, the following requirements define support for this architecture:
+
+- Architecture
+  - MUST deploy as a daemon on existing nodes, one Vector process per node.
+  - MUST deploy with Vector's [default agent configuration][default_agent_configuration]
+    which largely covers the agent architecture
+    [design recommendations][agent_architecture].
+- Sizing
+  - MUST deploy as a good infrastructure citizen, giving resource priority to
+    other services on the same node.
+  - SHOULD be limited to 2 vCPUs, MUST be overridable by the user.
+  - SHOULD be limited to 2 GiB of memory per vCPU (4 GiB in this case), MUST be
+    overridable by the user.
+  - SHOULD be limited to 1 GiB of disk space, MUST be overridable by the user.
+
+### 5. Aggregator Architecture
+
+The [aggregator architecture][aggregator_architecture] deploys Vector onto
+dedicated nodes for data aggregation. Along with general [hardening](#7-hardening)
+requirements, the following requirements define support for this architecture:
+
+- Architecture
+  - SHOULD deploy as a service on dedicated nodes, Vector is the only service
+    on the node.
```

Comment:
I guess this is a SHOULD for a VM target, but not for k8s?

Reply:
I think SHOULD is still reasonable - it ought to be done this way unless you understand the implications of not doing it, and have a good reason to ignore the directive. In kubernetes examples it could be binpacking/misc infra tools need to run there/etc.